pkg/sources/github_test.go

```diff
@@ -17,8 +18,13 @@ func TestGHSources_Get(t *testing.T) {
 	client := newGitHubClient(ctx, "")
 	sources := GitHub{Client: client}
 
+	// Require because the process modifies wd with chdir.
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+
 	t.Cleanup(func() {
-		_ = os.RemoveAll("./test")
+		_ = os.Chdir(wd)
+		_ = os.RemoveAll(filepath.Join(wd, "test"))
```

Comment:
Since `RemoveAll` is called with a complete path, would you please detail why you are calling `Chdir` first ?

Reply:
The `Chdir` is not related to `RemoveAll`.

```go
package plugin

import (
	"fmt"
	"os"
	"testing"
)

func TestA(t *testing.T) {
	os.Chdir("/")

	fmt.Println(os.Getwd())
}

func TestB(t *testing.T) {
	fmt.Println(os.Getwd())
}
```


```
=== RUN   TestA
/home/ldez/sources/traefik-labs/oss/plugins/piceus/internal/plugin <nil>
/ <nil>
--- PASS: TestA (0.00s)
=== RUN   TestB
/ <nil>
--- PASS: TestB (0.00s)
PASS

```

Reply:
I'm probably missing something. When removing this Chdir line, the two tests are green.
Would you please detail the regression that you identified ?

Reply:
The problem is not related to a green/red state.

The `Removal` is to fix the regression: the generated test files are not removed.

The `Chdir` is here to avoid side effects if other tests are created.